packages/core-components/src/layout/SignInPage/commonProvider.tsx

```diff
@@ -28,18 +28,21 @@ import { useApi, errorApiRef } from '@backstage/core-plugin-api';
 import { GridItem } from './styles';
 import { ForwardedError } from '@backstage/errors';
 import { UserIdentity } from './UserIdentity';
+import { PROVIDER_STORAGE_KEY } from './providers';
 
 const Component: ProviderComponent = ({ config, onSignInSuccess }) => {
-  const { apiRef, title, message } = config as SignInProviderConfig;
+  const { apiRef, title, message, id } = config as SignInProviderConfig;
   const authApi = useApi(apiRef);
   const errorApi = useApi(errorApiRef);
 
   const handleLogin = async () => {
     try {
+      localStorage.setItem(PROVIDER_STORAGE_KEY, id);
```

Comment:
Moving this to the provider implementations here is actually a bit awkward. It'll be much better if we can keep all of the local storage logic inside `providers.tsx`. I think the cleanest solution is to add a `onSignInStarted` callback or something like that, that allows us to set the selected provider.

Using this structure we'd need to add it to the custom and guest providers too.

Reply:
I will also add a simple `onSignInFailure` to remove the provider id from local storage.